task3/graph.c

```diff
@@ -0,0 +1,163 @@
+    Node *node = calloc(1, sizeof(Node));
+    if (node == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    node->previous = previous;
+    node->vertex = vertex;
+    return node;
+}
+
+Graph buildGraph(const char *filePath, bool *errorCode) {
+    FILE *file = fopen(filePath, "r");
+    if (file == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    int verticesAmount = 0;
+    fscanf(file, "%d", &verticesAmount);
+    Graph graph = createGraph(verticesAmount, errorCode);
+    if (graph == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    for (int i = 0; i < verticesAmount; ++i) {
+        for (int j = 0; j < verticesAmount; ++j) {
+            int edge = 0;
+            fscanf(file, "%d", &edge);
+            if (edge != 0) {
+                graph->adjacencyMatrix[i][j] = 1;
+                graph->vertices[i]->linkedVertices = addNode(graph->vertices[i]->linkedVertices, graph->vertices[j], errorCode);
+            }
+        }
+    }
+    return graph;
+}
+
+
+Vertex *createVertex(Value number, bool *errorCode) {
+    Vertex *vertex = calloc(1, sizeof(Vertex));
+    if (vertex == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    vertex->number = number;
+    vertex->linkedVertices = NULL;
+    return vertex;
+}
+
+Graph createGraph(const int verticesAmount, bool *errorCode) {
+    Graph graph = calloc(1, sizeof(struct Graph));
+    if (graph == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    graph->verticesAmount = verticesAmount;
+
+    graph->adjacencyMatrix = calloc(verticesAmount, sizeof(int*));
+    if (graph->adjacencyMatrix == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    for (int i = 0; i < verticesAmount; ++i) {
+        graph->adjacencyMatrix[i] = calloc(verticesAmount, sizeof(int));
+        if (graph->adjacencyMatrix[i] == NULL) {
+            *errorCode = true;
+            return NULL;
+        }
+    }
+
+    graph->vertices = calloc(verticesAmount, sizeof(Vertex));
+    if (graph->vertices == NULL) {
+        *errorCode = true;
+        return NULL;
+    }
+    for (int i = 0; i < verticesAmount; ++i) {
+        graph->vertices[i] = createVertex(i, errorCode);
+        if (*errorCode) {
+            return NULL;
+        }
+    }
+    return graph;
+}
+
+void doWidthTraversal(Graph graph, bool *isVertexVisited, const int startingVertex, bool *errorCode) {
+    Queue queue = createQueue();
+    if (queue == NULL) {
+        *errorCode = true;
+        return;
+    }
+
+    enqueue(queue, graph->vertices[startingVertex]);
+    while (!isQueueEmpty(queue)) {
+        Vertex *currentVertex = dequeue(queue);
+        printf("%d ", currentVertex->number);
+        isVertexVisited[currentVertex->number] = true;
+        Node *linkedVertices = currentVertex->linkedVertices;
+        while(linkedVertices != NULL) {
+            if (isVertexVisited[linkedVertices->vertex->number] == false) {
```

Comment:
Стайлгайд запрещает сравниваться с булевыми константами
```suggestion
            if (!isVertexVisited[linkedVertices->vertex->number]) {
```